app/survey/components/graphs/WhyFairhold.tsx

```diff
@@ -5,7 +5,12 @@ import { Bar, BarChart, XAxis, YAxis, ResponsiveContainer } from "recharts";
 import { useSurveyContext } from "../../context";
 
 export const WhyFairhold = () => {
-  const whyFairhold = useSurveyContext().barOrPie.whyFairhold;
+  let { whyFairhold } = useSurveyContext().barOrPie;
+  whyFairhold = whyFairhold.slice(0,5);
+  const maxValue = whyFairhold.length > 0 ? whyFairhold[0].value : 0;
+  const maxResponses = Math.ceil(maxValue / 10) * 10;
+  const xDomainMax = maxResponses < 10 ? 10 : maxResponses;
```

Comment:
This should be wrapped in a well-named function - without context it's hard to know what's going on here - it's also a repeated bit of logic.

Maybe something like this?

```ts

const getTopFive = (data: BarOrPieResult[]) => data.slice(0,5);

const calculateChartMaximum = (data: BarOrPieResult[]) => {
  const maxValue = data[0]?.value ?? 0;
  const maxResponses = Math.ceil(maxValue / 10) * 10;
  const chartMax = maxResponses < 10 ? 10 : maxResponses;
  return chartMax
}

```